modules/auxiliary/scanner/ssh/ssh_login_pubkey.rb

```diff
@@ -201,20 +204,30 @@ def realm
       nil
     end
 
-    def valid!
+    def valid?
+      @error_list = []
       @key_data = Set.new
       if File.directory?(@key_path)
         @key_files ||= Dir.entries(@key_path).reject { |f| f =~ /^\x2e|\x2epub$/ }
         @key_files.each do |f|
-          data = read_key(File.join(@key_path, f))
-          @key_data << data if valid_key?(data)
+          begin
+            data = read_key(File.join(@key_path, f))
+            @key_data << data if valid_key?(data)
+          rescue StandardError => e
```

Comment:
Would `IOError` not make more sense here? `StandardError` is the default and if you just put a `rescue` clause by itself, then it will default to `StandardError`. As a result, it captures a rather large swath of exceptions.

Reply:
I believe `IOError` won't work as that was the error being thrown before the fix was implemented. 
I intended to change `StandError` too `OpenSSL::PKey::PKeyError`, as that is the new error is you try to read a file that isn't a valid key. I had `StandError` in for my tests to verify there was any other errors being thrown, but I forgot to get that swapped back over.

I think that adding `OpenSSL::PKey::PKeyError` in instead of `StandError` would be the right change here, but let me know what you think.

Thanks for pointing this out 👍

Reply:
I think sticking with StandardError makes sense to me, as we don't want an edge case in loading a particular file to block the user from running this module

Reply:
Additional perspective:

It seemed like there was two potential outcomes here for this code suggestion:
1. We replace `StandardError` with only `IOError`, and we forget about an edge case exception that is absolutely fine to ignore and report the issue to the issue. Unfortunately since we don't catch it, the exception causes the module to break and not continue processing the remainding files
2. We stick with 'StandardError' and unexpected edge cases are safely ignored. There's the potential of certain exceptions being caught and logged to the user which might require an additional code fix - however this wouldn't impact the user, and the the module can still safely be used.

For point 1, had we only gone with `IOError`, it wouldn't have caught the possibility of a raised `OpenSSL::PKey::PKeyError` exception - and thus would have impacted a user from running this module.

That's why I think i this case sticking with `StandardError` seems like an okay decision to make in this scenario, particularly as it's wrapped around such a small surface area of code.

Reply:
Sounds good, and thanks for the perspective on the `OpenSSL:PKey::PKeyError` exception. Will keep this as a `StandardError` capture then.